src/dynamic_reconfigure/parameter_generator_catkin.py

```diff
@@ -581,11 +581,13 @@ def _rreplace_str_with_val_in_dict(self, orig_dict, old_str, new_val):
             if isinstance(v, dict):
                 self._rreplace_str_with_val_in_dict(v, old_str, new_val)
             elif isinstance(v, list):
+                idx = 0
```

Comment:
You can use `enumerate()` here, to save yourself a line. It would look like this:

```python
for idx, i in enumerate(v):
    if isinstance(i, str) and i == old_str:
        orig_dict[k][idx] = new_val
    elif isinstance(i, dict):
        self._rreplace_str_with_val_in_dict(i, old_str, new_val)
```